mobius-extras/src/main/java/com/spotify/mobius/extras/Function.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.extras;
+
+import javax.annotation.Nullable;
+
+/** Interface for simple functions that can return null. */
+public interface Function<T, R> {
```

Comment:
Should be called something else to avoid confusion with `com.spotify.mobius.functions.Function`. Maybe `NullValuedFunction` or something similar?

Reply:
yeah I initially had that but it didn't feel very elegant tbh.. I made the assumption that most people will be passing in lambdas so the interface name won't really matter in that case. Do you want me to change it still?

Reply:
Hm, it's true people will generally use lambdas. I tried cmd-P in my IntelliJ, which gives me information about parameters with my key bindings. That just gives the unqualified class name, so it would say `Function` in this case, and would look just like the non-nullable version. I guess that means it's more important that the name is clear when you look at it, but doesn't matter so much if it is inelegant because it won't show up in code thanks to lambdas. I kind of think for me it would be easier to get it right with a more explicit name, especially since it's normally hidden. But I don't have a very strong opinion because not everyone would do it the same way I do. You make the call.